yapapi/executor/__init__.py

```diff
@@ -206,14 +220,17 @@ async def submit(
                     logger.debug("Cancelling task: %s", task)
                     task.cancel()
             await asyncio.gather(*all_tasks, return_exceptions=True)
+            # Signal that this computation is finished
+            computation_finished.set()
+            self._active_computations.remove(computation_finished)
 
     async def _submit(
         self,
         worker: Callable[[WorkContext, AsyncIterator[Task[D, R]]], AsyncGenerator[Work, None]],
         data: Iterable[Task[D, R]],
         services: Set[asyncio.Task],
         workers: Set[asyncio.Task],
-    ) -> AsyncIterator[Task[D, R]]:
+    ) -> AsyncGenerator[Task[D, R], None]:
```

Comment:
We have to give a more specific type here, to be able to call `athrow` on an object returned from `_submit()` without `mypy` protesting.